pkg/controller/podgroup_utils.go

```diff
@@ -0,0 +1,73 @@
+package controller
+
+import (
+	"fmt"
+	"github.com/kubernetes-sigs/kube-batch/pkg/apis/scheduling/v1alpha1"
+	v1 "github.com/microsoft/frameworkcontroller/pkg/apis/frameworkcontroller/v1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/klog"
+)
+// todo frameworkController add podgroupInformer and save to cache
+func (c *FrameworkController) SyncPodGroup(f *v1.Framework, minAvailableReplicas int32) (*v1alpha1.PodGroup, error) {
+
+	kubeBatchClientInterface := c.kbClient
+	// Check whether podGroup exists or not
+	podGroupName := GenPodGroupName(f.Name)
+	podGroup, err := kubeBatchClientInterface.SchedulingV1alpha1().PodGroups(f.Namespace).Get(podGroupName, metav1.GetOptions{})
+	if err == nil {
+		return podGroup, nil
+	}
+
+	// create podGroup with
+	minAvailable := intstr.FromInt(int(minAvailableReplicas))
+	priorityClassName := f.Spec.PriorityClassName
+	queueName := f.Spec.Queue
+	klog.Infof("create podgroup [%s/%s] minNumber %v, PriorityClassName %v, Queue %v", f.Namespace, podGroupName, priorityClassName, queueName)
+	createPodGroup := &v1alpha1.PodGroup{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      podGroupName,
+			Namespace: f.Namespace,
+		},
+		Spec: v1alpha1.PodGroupSpec{
+			MinMember:         minAvailable.IntVal,
+			PriorityClassName: priorityClassName,
+			Queue:             queueName,
+		},
+	}
+	return kubeBatchClientInterface.SchedulingV1alpha1().PodGroups(f.Namespace).Create(createPodGroup)
+}
+func (jc *FrameworkController) DeletePodGroup(object runtime.Object) error {
+	kubeBatchClientInterface := jc.kbClient
```

Comment:
Why not make framework as the owner of its pod group? So you do not need to manage it.
See [https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/)

Reply:
thanks, i'll add it.